service_grpc_flight/src/lib.rs

```diff
@@ -332,131 +450,116 @@ where
         Ok(tonic::Response::new(flight_info))
     }
 
     async fn do_put(
         &self,
         _request: Request<Streaming<FlightData>>,
     ) -> Result<Response<Self::DoPutStream>, tonic::Status> {
-        Err(tonic::Status::unimplemented("Not yet implemented"))
+        Err(tonic::Status::unimplemented("Not yet implemented: do_put"))
     }
 
     async fn do_action(
         &self,
         _request: Request<Action>,
     ) -> Result<Response<Self::DoActionStream>, tonic::Status> {
-        Err(tonic::Status::unimplemented("Not yet implemented"))
+        Err(tonic::Status::unimplemented(
+            "Not yet implemented: do_action",
+        ))
     }
 
     async fn list_actions(
         &self,
         _request: Request<Empty>,
     ) -> Result<Response<Self::ListActionsStream>, tonic::Status> {
-        Err(tonic::Status::unimplemented("Not yet implemented"))
+        Err(tonic::Status::unimplemented(
+            "Not yet implemented: list_actions",
+        ))
     }
 
     async fn do_exchange(
         &self,
         _request: Request<Streaming<FlightData>>,
     ) -> Result<Response<Self::DoExchangeStream>, tonic::Status> {
-        Err(tonic::Status::unimplemented("Not yet implemented"))
+        Err(tonic::Status::unimplemented(
+            "Not yet implemented: do_exchange",
+        ))
     }
 }
 
-impl<S> FlightService<S>
-where
-    S: QueryNamespaceProvider,
-{
-    /// Given a successfully decoded protobuf *Any* message, handles
-    /// recognized messages (e.g those defined by FlightSQL) and
-    /// creates the appropriate FlightData response
-    ///
-    /// Arguments
-    ///
-    /// namespace_name: is the target namespace of the request
-    ///
-    /// flight_descriptor: is the descriptor sent in the request (included in response)
-    ///
-    /// msg is the `cmd` field of the flight descriptor decoded as a protobuf  message
-    async fn dispatch(
-        &self,
-        namespace_name: &str,
-        flight_descriptor: FlightDescriptor,
-        msg: prost_types::Any,
-    ) -> Result<FlightInfo> {
-        fn try_unpack<T: ProstMessageExt>(msg: &prost_types::Any) -> Result<Option<T>> {
-            // Does the type URL match?
-            if T::type_url() != msg.type_url {
-                return Ok(None);
-            }
-            // type matched, so try and decode
-            let m = prost::Message::decode(&*msg.value).context(DeserializationTypeKnownSnafu {
-                type_url: &msg.type_url,
-            })?;
-            Ok(Some(m))
+/// Extracts an encoded Protobuf message from a [`FlightDescriptor`],
+/// as used in FlightSQL.
+fn msg_from_descriptor(flight_descriptor: FlightDescriptor) -> Result<prost_types::Any> {
+    match flight_descriptor.r#type() {
+        DescriptorType::Cmd => {
+            let msg: prost_types::Any =
+                Message::decode(flight_descriptor.cmd.as_ref()).context(DeserializationSnafu)?;
```

Comment:
```suggestion
                Message::decode(flight_descriptor.cmd).context(DeserializationSnafu)?;
```
This ensures that zero copy works correctly by passing in `Bytes` instead of `&[u8]`

Reply:
in b8f761cf5